File: src/main/java/com/example/inventory/query/ProductQueryEntity.java
package com.example.inventory.query;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "products")
public class ProductQueryEntity {

    @Id
    private String productId;
    private String name;
    private int quantity;

    protected ProductQueryEntity() {
    }

    public ProductQueryEntity(String productId, String name, int quantity) {
        this.productId = productId;
        this.name = name;
        this.quantity = quantity;
    }

    public String getProductId() {
        return productId;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}

File: src/main/kotlin/com/example/inventory/query/ProductQueryEntity.kt
package com.example.inventory.query

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "products")
class ProductQueryEntity() {

    @Id
    var productId: String = ""
    var name: String = ""
    var quantity: Int = 0

    constructor(productId: String, name: String, quantity: Int) : this() {
        this.productId = productId
        this.name = name
        this.quantity = quantity
    }
}
